nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -0,0 +1,504 @@
+package deploymentwatcher
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+const (
+	// evalBatchPeriod is the batching length before creating an evaluation to
+	// trigger the scheduler when allocations are marked as healthy.
+	evalBatchPeriod = 1 * time.Second
```

Comment:
I would give this a clearer name since there is two variables with almost the same name. This is the per-job batching period